### ISSUE TITLE
Problem with InfluxdbHandler

### ISSUE THREAD

**Issue by csarcom** [python-diamond/Diamond#137]
I have installed diamond and influxdb, both running, my diamonf conf setup is:

``` python
# Handlers for published metrics.
handlers = diamond.handler.graphite.GraphiteHandler, diamond.handler.archive.ArchiveHandler, diamond.handler.influxdbHandler.InfluxdbHandler

[[InfluxdbHandler]]
### Options for InfluxdbHandler
hostname = localhost
database = graphite
batch_size = 1
username = diamond
password = diamond
ssl = False
port = 8086

# Default Poll Interval (seconds)
interval = 10
```

The rest is all default conf.

I Have no problem accessing influxDB web interface and create databases and do some post... But for some reason diamond is not writing any data to infliuxDB, Am I missing something?

**Comment by csarcom** [comment]
I changed influxdb version to 0.4.1 and now its working.

I was getting this error:

```python
1427131390.31	[Handlers:25733:ERROR]	Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/diamond/handler/Handler.py", line 72, in _process
    self.process(metric)
  File "/usr/lib/python2.7/site-packages/diamond/handler/influxdbHandler.py", line 151, in process
    self._send()
  File "/usr/lib/python2.7/site-packages/diamond/handler/influxdbHandler.py", line 182, in _send
    time_precision=self.time_precision)
  File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 242, in write_points
    return self._write_points(data=data, time_precision=time_precision)
  File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 280, in _write_points
    expected_response_code=200
  File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 185, in request
    raise InfluxDBClientError(response.content, response.status_code)
InfluxDBClientError: 404: 404 page not found
```

**Comment by ctrlrsf** [comment]
python influxdb client version 1.0.0 is targeting v0.9 API, which has breaking changes from 0.8.x API.

From influxdb-python project:
InfluxDB 0.8.X users may use the legacy client by using `from influxdb.influxdb08 import InfluxDBClient` instead.

What are the thoughts on adding support for v0.9, sticking with 0.8.x for a while, or supporting both?

**Comment by benjarrell** [comment]
I would like both for now.

Using 1.0.2 with a v0.9.0-rc17 database I get:

```python
1427820114.45  [Handlers:4771:DEBUG]   InfluxdbHandler: Sending batch sizeof : 93/10 after 4.018292s
1427820114.45  [Handlers:4771:DEBUG]   InfluxdbHandler: Socket is not connected. Reconnecting.
1427820114.45  [Handlers:4771:DEBUG]   InfluxdbHandler: Established connection to 1.2.3.4:8086/diamond.
1427820114.45  [Handlers:4771:DEBUG]   InfluxdbHandler: writing 93 series of data
1427820114.46  [Handlers:4771:DEBUG]   InfluxdbHandler: Error sending metrics, waiting for 8s.
1427820114.46  [Handlers:4771:ERROR]   Traceback (most recent call last):
File "/usr/lib/python2.7/site-packages/diamond/handler/Handler.py", line 72, in _process
  self.process(metric)
File "/usr/lib/python2.7/site-packages/diamond/handler/influxdbHandler.py", line 151, in process
  self._send()
File "/usr/lib/python2.7/site-packages/diamond/handler/influxdbHandler.py", line 182, in _send
  time_precision=self.time_precision)
File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 256, in write_points
  retention_policy=retention_policy)
File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 290, in _write_points
  expected_response_code=200
File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 200, in write
  expected_response_code=expected_response_code
File "/usr/lib/python2.7/site-packages/influxdb/client.py", line 191, in request
  raise InfluxDBClientError(response.content, response.status_code)
InfluxDBClientError: 500: {"error":"fields required"}
1427820114.46  [Handlers:4771:ERROR]   /usr/lib/python2.7/site-packages/diamond/handler/Handler.py:74
```

**Comment by imreACTmd** [comment]
I just sent in a pull request to support both 0.8 and 0.9.

**Comment by rhunwicks** [comment]
In case anyone else is trying to follow - the PR is #148

**Comment by shortdudey123** [comment]
Merged #504.
